plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1523,14 +1529,7 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
                   }
                };
 
-               auto itr = unapplied_trxs.begin();
-               while( itr != unapplied_trxs.end() ) {
-                  auto itr_next = itr; // save off next since itr may be invalidated by loop
-                  ++itr_next;
-
-                  if( preprocess_deadline <= fc::time_point::now() ) exhausted = true;
-                  if( exhausted ) break;
```

Comment:
Because `next()` returns a popped trx that needs to be processed. If you `break` after the `next()` call you would drop the trx.

Reply:
Previously if `exhausted` was set to true by the previous stage dealing with persisted transactions, we would break out of the while loop in the first iteration before doing anything else, and then print the log "Processed ${m} of ${n} previously applied transactions...".

Now it will do additional work like remove the first transaction in the unapplied transaction queue and maybe try to apply it.

So I think the while loop should be guarded by a check to ensure that `exhausted` is false.

Reply:
Done.